src/madness/chem/potentialmanager.h

```diff
@@ -106,6 +106,49 @@ class CoreOrbitalDerivativeFunctor : public FunctionFunctorInterface<double,3> {
     };
 };
 
+class NuclearDensityFunctor : public FunctionFunctorInterface<double,3> {
```

Comment:
I understand how the existing code isn't supporting non-periodic dimensions properly, and why we'd need multiple R. I don't understand the problem with the first point. My understanding was that `maxR` was needed for adding periodic images on the boundary. The plan (which I'll need to revise because _only some dimensions may be periodic_) was that the `maxR = 0` constructor was for the non-periodic case, and `maxR = 1` was for periodic. Is there some reason why I want `maxR = 1` even in the case of OBC?

Reply:
Note that current version still doesn't support non-periodic dimensions properly. `maxR` needs to support being different for different dimensions. I can deal with that later - right now, the MPQC side of the problem needs attention more.

Reply:
ignore your branch, I pushed all changes into new branch in main repo ...

Reply:
see https://github.com/m-a-d-n-e-s-s/madness/compare/jm/cleanup/nuclear-density-functor?expand=1#diff-32ce43f9e9a98e2973ffcf2e02717a3898ad9c43cf28bc2d9e435ba09ea77533R52